app/gui/view/src/project.rs

```diff
@@ -26,6 +27,45 @@ use ensogl_hardcoded_theme::Theme;
 
 
 
+// =====================
+// === SearcherInput ===
+// =====================
+
+/// The information needed to setup Searcher Controller.
+#[derive(Clone, CloneRef, Copy, Debug, PartialEq)]
+pub enum SearcherInput {
```

Comment:
I can't think of a better name and/or location for this enum. It is used both for created nodes and for edited nodes (all cases when the searcher is opened), and it can't be placed in controllers, as it must be used in the View's FRP.

Reply:
I think the placement is ok, if we change the meaning of this structure a little: not "The information needed to set up the Searcher Controller." but rather "An enum describing how the searcher was opened". 

And adjust the name to that: this is not actually the searcher input, because the edge is not a part of edge input (and is a part of one variant of this enum). `WayOfOpeningSearcher` is better, or something more compact if you find.